x-pack/platform/plugins/shared/triggers_actions_ui/public/application/components/json_editor_with_message_variables.tsx

```diff
@@ -77,7 +77,7 @@ export const JsonEditorWithMessageVariables: React.FunctionComponent<Props> = ({
   const { convertToJson, setXJson, xJson } = useXJsonMode(inputTargetValue ?? null);
 
   useEffect(() => {
-    if (!xJson && inputTargetValue) {
+    if (inputTargetValue && (!xJson || xJson !== inputTargetValue)) {
```

Comment:
Instead of changing this `useEffect` what about using the `key` prop in `JsonEditorWithMessageVariables` component?

For example:

```
<JsonEditorWithMessageVariables key={subActionsParams.template} />
```